### PR TITLE
Add new remote params

### PR DESCRIPTION
**Pull request by Jeasmine**
* Add locationShared remote param
   * Add requiresUserPrivacyConcent remote param
   * Keep public method
   * Add tests for new getters

---
This change is [Reviewable](https://reviewable.io/reviews/onesignal/onesignal-ios-sdk/707)

### CODE REVIEW

File: iOS_SDK/OneSignalSDK/Source/OneSignal.m [hunk 1]
```diff
@@ -161,13 +160,12 @@ @implementation OneSignal
...
 static BOOL promptBeforeOpeningPushURLs = false;
 
-
-// Indicates if initialization of the SDK has been delayed until the user gives privacy consent
-static BOOL delayedInitializationForPrivacyConsent = false;
-
 // If initialization is delayed, this object holds params such as the app ID so that the init()
 // method can be called the moment the user provides privacy consent.
-DelayedInitializationParameters *delayedInitParameters;
+DelayedInitializationParameters *_delayedInitParameters;
```
Comment (mikechoch) [inline iOS_SDK/OneSignalSDK/Source/OneSignal.m:165]: Lets keep this consistent with Android SDK class `DelayedConsentInitializationParameters`

File: iOS_SDK/OneSignalSDK/Source/OneSignalTracker.m [hunk 2]
```diff
@@ -81,6 +82,10 @@ + (void)endBackgroundFocusTask {
 }
 
 + (void)onFocus:(BOOL)toBackground {
+    // Privacy consent can be set by remote params, need to wait until they are set
```
Comment (mikechoch) [inline iOS_SDK/OneSignalSDK/Source/OneSignalTracker.m:85]: Simplify comment?
`// Privacy consent is set through remote params request`

File: iOS_SDK/OneSignalSDK/Source/OneSignalTracker.m [hunk 3]
```diff
@@ -81,6 +82,10 @@ + (void)endBackgroundFocusTask {
...
+    if (![OneSignal downloadedParameters])
```
Comment (mikechoch) [inline iOS_SDK/OneSignalSDK/Source/OneSignalTracker.m:86]: We deleted this boolean on the Android SDK can we do the same for iOS?

File: iOS_SDK/OneSignalSDK/UnitTests/Shadows/OneSignalClientOverrider.h [hunk 4]
```diff
@@ -48,5 +48,7 @@
 +(void)disableExecuteRequestOverride:(BOOL)disable;
 +(NSArray<OneSignalRequest *> *)executedRequests;
 + (void)setMockResponseForRequest:(NSString *)request withResponse:(NSDictionary *)response;
++ (NSDictionary*)iosParamsResponse;
```
Comment (mikechoch) [inline iOS_SDK/OneSignalSDK/UnitTests/Shadows/OneSignalClientOverrider.h:51]: We are using iOS SDK so I think we can remove ios and just call this 
`remoteParamsResponse`?
`remoteParams`?

File: iOS_SDK/OneSignalSDK/UnitTests/Shadows/OneSignalClientOverrider.h [hunk 5]
```diff
@@ -48,5 +48,7 @@
...
++ (void)setRemoteParamResponse:(NSDictionary *)params;
```
Comment (mikechoch) [inline iOS_SDK/OneSignalSDK/UnitTests/Shadows/OneSignalClientOverrider.h:52]: `setRemoteParamResponse` should have `params` in plural
`setRemoteParamsResponse`?

File: iOS_SDK/OneSignalSDK/UnitTests/Shadows/OneSignalClientOverrider.m [hunk 6]
```diff
@@ -73,12 +74,15 @@ + (void)load {
...
 + (NSDictionary*)iosParamsResponse {
-    return @{
-        @"fba": @true,
-        IOS_REQUIRES_EMAIL_AUTHENTICATION : @(requiresEmailAuth),
-        IOS_USES_PROVISIONAL_AUTHORIZATION : @(shouldUseProvisionalAuthorization),
-        OUTCOMES_PARAM : iOSParamsOutcomes
-    };
+    return remoteParams ? remoteParams :
+        @{
+            @"fba": @true,
```
Comment (mikechoch) [inline iOS_SDK/OneSignalSDK/UnitTests/Shadows/OneSignalClientOverrider.m:79]: Add a `NSString` for this `@"fba"` similar to the params below it

File: iOS_SDK/OneSignalSDK/UnitTests/Shadows/OneSignalClientOverrider.m [hunk 7]
```diff
@@ -55,6 +55,7 @@ @implementation OneSignalClientOverrider
 static NSMutableArray<OneSignalRequest *> *executedRequests;
 static NSMutableDictionary<NSString *, NSDictionary *> *mockResponses;
 static NSDictionary* iOSParamsOutcomes;
```
Comment (mikechoch) [inline iOS_SDK/OneSignalSDK/UnitTests/Shadows/OneSignalClientOverrider.m:57]: Remove ios and rename?
`remoteParamsOutcomes`?

File: iOS_SDK/OneSignalSDK/Source/OneSignalTracker.m (hunk 2 above)
Comment (Jeasmine) [reply iOS_SDK/OneSignalSDK/Source/OneSignalTracker.m]: given that we need the delay task to have this check, I removed it

File: iOS_SDK/OneSignalSDK/Source/OneSignalTracker.m (hunk 3 above)
Comment (Jeasmine) [reply iOS_SDK/OneSignalSDK/Source/OneSignalTracker.m]: given that we need the delay task to have this check, I removed it, will need to check if this is need tho

File: iOS_SDK/OneSignalSDK/UnitTests/Shadows/OneSignalClientOverrider.h (hunk 4 above)
Comment (Jeasmine) [reply iOS_SDK/OneSignalSDK/UnitTests/Shadows/OneSignalClientOverrider.h]: remoteParamsResponse 👍

File: iOS_SDK/OneSignalSDK/Source/OneSignal.m [hunk 8]
```diff
@@ -1408,7 +1411,7 @@ + (void)promptLocationFallbackToSettings:(BOOL)fallback completionHandler:(void
 }
 
 + (BOOL)isLocationShared {
-    return mShareLocation;
+    return [OneSignalUserDefaults.initShared getSavedBoolForKey:OSUD_LOCATION_ENABLED defaultValue:YES];;
```
Comment (mikechoch) [inline iOS_SDK/OneSignalSDK/Source/OneSignal.m:1414]: Extra `;` at end

Comment (mikechoch) [inline iOS_SDK/OneSignalSDK/Source/OneSignal.m:1414]: Extra `;` at end

File: iOS_SDK/OneSignalSDK/Source/OneSignal.m [hunk 9]
```diff
@@ -808,6 +806,12 @@ + (void)downloadIOSParamsWithAppId:(NSString *)appId {
 
         if (result[IOS_RECEIVE_RECEIPTS_ENABLE] != (id)[NSNull null])
             [OneSignalUserDefaults.initShared saveBoolForKey:OSUD_RECEIVE_RECEIPTS_ENABLED withValue:[result[IOS_RECEIVE_RECEIPTS_ENABLE] boolValue]];
+        
+        if (result[IOS_LOCATION_SHARED])
```
Comment (jkasten2) [inline iOS_SDK/OneSignalSDK/Source/OneSignal.m:810]: This doesn't match the logic we have for Android. We need to check for key presence here.
See this commit on Android:
https://github.com/OneSignal/OneSignal-Android-SDK/pull/1124/files#diff-8f1d87ed84bc5cc2e615dcfb826df87bR2522

And this comment for the remote param override logic we want for fallback support.
https://github.com/OneSignal/OneSignal-Android-SDK/pull/1124#pullrequestreview-477112796

Comment (Jeasmine) [reply iOS_SDK/OneSignalSDK/Source/OneSignal.m]: Added matched logic with Android, we will need the delayed task controller to match 100% the logic, on iOS we are not waiting for remote params due to this missing behaviour

**Review by jkasten2 — APPROVED** [review]

**Pull request merged**